src/json.c

```diff
@@ -101,17 +62,98 @@ int FieldSpec_CheckJsonType(FieldType fieldType, JSONType type) {
     break;
   // Boolean values can be represented only as TAG 
   case JSONType_Bool:    
-  // An array can be represented only as TAG 
+  // An array can be represented only as TAG
+  case JSONType_Array:
     if (fieldType == INDEXFLD_T_TAG) {
       rv = REDISMODULE_OK;
     }
     break;
   // An object or null type are not supported
-  case JSONType_Array: // TODO: array will be supported as TAG
   case JSONType_Object:
   case JSONType_Null:
   case JSONType__EOF:
     break;
   }
   return rv;
-}
\ No newline at end of file
+}
+
+int JSON_GetRedisModuleString(RedisModuleCtx *ctx, JSONResultsIterator jsonIter,
+                              FieldType ftype, struct DocumentField *df) {
+  RedisJSON json = japi->next(jsonIter);
```

Comment:
You are only getting the first reply so we are back to where we were before the API changes. You should iterate on all the replies and take them ...